content/blog/aws-lambda-efs/index.md

```diff
@@ -0,0 +1,228 @@
+---
+title: "Using AWS Lambda with Amazon Elastic File Service (EFS)"
+date: 2020-06-16
+draft: true
+meta_desc: "Pulumi supports Amazon EFS with Lambda and Fargate services."
+meta_image: aws-lambda-efs.png
+authors:
+    - luke-hoban
+tags:
+    - AWS
+    - Lambda
+    - EFS
+---
+
+Ever since AWS Lambda was released in 2015, users have wanted persistent file storage beyond the small 512MB `/tmp` disk allocated to each Lambda function.  The following year, Amazon launched EFS,  offering a simple managed file system service for AWS, but initially only available to mount onto Amazon EC2 instances. Over the last few months, AWS has been extending access to EFS to all of the modern compute offerings.  First [EKS](https://aws.amazon.com/about-aws/whats-new/2019/09/amazon-eks-announces-beta-release-of-amazon-efs-csi-driver/) for Kubernetes, then [ECS and Fargate](https://aws.amazon.com/about-aws/whats-new/2020/04/amazon-ecs-aws-fargate-support-amazon-efs-filesystems-generally-available/) for containers.  Today, AWS announced that [EFS is now also supported in Lambda](TBD), providing easy access to network file systems from your serverless functions.
+
+<!--more-->
+
+In this post, we’ll show you how to get started using EFS with Lambda and Pulumi.
+
+Why would you *want* to use EFS from Lambda?  There are a few common reasons:
+
+1. You need more local disk than the 512MB of `/tmp` space available to a single Lambda Function.
+1. You want to share persistent storage across various functions and would prefer to use file system storage instead of blob storage or a database.
+1. You want to share persistent storage between Lambda and other compute (like containers or ECS), and would prefer to use file storage over blob storage or a database.
+
+Storing data to the file system is particularly natural for the kind of small functions common in Lambda - no need to use the AWS SDK - just call `fs.writeFileSync()` (or equivalent for your language of choice).
+
+## EFS + Lambda Concepts
+
+To use an EFS file system from Lambda, we need a few things:
+
+1. A VPC
+1. An EFS File System in the VPC
+1. EFS mount points in each of the subnets of our VPC
+1. An EFS Access Point specifying how Lambda should mount the EFS volume
+1. An AWS Lambda Function running in the same VPC and private subnets and referencing the EFS Access Point
+
+![Lambda with EFS Architecture](lambdaefs.png)
+
+Let's walk through how to build the required infrastructure.  We'll use Pulumi to provision the necessary resources.  You can check out the [full source code](https://github.com/pulumi/examples/tree/master/aws-ts-lambda-efs) in the Pulumi Examples.
+
+> Note: These examples depend on the newly added support for EFS with Lambda in the `2.10.0` release of the `pulumi-aws` package, which will be released later today.
```

Comment:
I think we can nuke this note, esp. since "today" is (hopefully) now meaningless!

Reply:
Yeah - this is available now - so we can remove the note now.